addons/web/static/src/js/views/graph/graph_owl_renderer.js

```diff
@@ -0,0 +1,947 @@
+            return COLORS[index % COLOR_NB];
+        }
+
+        /**
+         * Prepare options for the chart according to the current mode (= chart type).
+         * This function returns the parameter options used to instantiate the chart
+         *
+         * @private
+         * @param {number} datasetsCount
+         * @returns {Object} the chart options used for the current mode
+         */
+        _prepareOptions(datasetsCount) {
+            return {
+                maintainAspectRatio: false,
+                scales: this._getScaleOptions(),
+                legend: this._getLegendOptions(datasetsCount),
+                tooltips: this._getTooltipOptions(),
+                elements: this._getElementOptions(),
+            };
+        }
+
+        /**
+         * Returns an object used to style chart elements independently from the datasets.
+         *
+         * @private
+         * @returns {Object}
+         */
+        _getElementOptions() {
+            const elementOptions = {};
+            if (this.props.mode === 'bar') {
+                elementOptions.rectangle = {borderWidth: 1};
+            } else if (this.props.mode === 'line') {
+                elementOptions.line = {
+                    tension: 0,
+                    fill: false,
+                };
+            }
+            return elementOptions;
+        }
+
+        /**
+         * Returns the options used to generate chart tooltips.
+         *
+         * @private
+         * @returns {Object}
+         */
+        _getTooltipOptions() {
+            const tooltipOptions = {
+                // disable Chart.js tooltips
+                enabled: false,
+                custom: this._customTooltip.bind(this),
+            };
+            if (this.props.mode === 'line') {
+                tooltipOptions.mode = 'index';
+                tooltipOptions.intersect = false;
+            }
+            return tooltipOptions;
+        }
+
+        /**
+         * This function creates a custom HTML tooltip.
+         *
+         * @private
+         * @param {Object} tooltipModel see chartjs documentation
+         */
+        _customTooltip(tooltipModel) {
+            if (this.tooltip) {
+                this.tooltip.remove();
+            }
+
+            if (tooltipModel.opacity === 0) {
+                return;
+            }
+            if (tooltipModel.dataPoints.length === 0) {
+                return;
+            }
+
+            const chartArea = this.chart.chartArea;
+            const chartAreaLeft = chartArea.left;
+            const chartAreaRight = chartArea.right;
+            const chartAreaTop = chartArea.top;
+            const rendererTop = this.el.getBoundingClientRect().top;
+
+            const maxTooltipLabelWidth = Math.floor((chartAreaRight - chartAreaLeft) / 1.68) + 'px';
+            const tooltipItems = this._getTooltipItems(tooltipModel);
+
+            const template = document.createElement('template');
```

Comment:
why 'template'?

Reply:
Following the advices here: https://stackoverflow.com/a/35385518/1824793

> Note that similar approaches that use a different container element such as a div don't quite work. HTML has restrictions on what element types are allowed to exist inside which other element types; for instance, you can't put a td as a direct child of a div. This causes these elements to vanish if you try to set the innerHTML of a div to contain them. Since <template>s have no such restrictions on their content, this shortcoming doesn't apply when using a template.